sympy/solvers/solvers.py

```diff
@@ -1136,6 +1137,28 @@ def _has_piecewise(e):
         if _has_piecewise(fi):
             f[i] = piecewise_fold(fi)
 
+    # expand double angles; in general, expand_trig will allow
+    # more roots to be found but this is not a great solultion
+    # to not returning a parametric solution, otherwise
+    # many values can be returned that have a simple
+    # relationship between values
+    targs = {t for fi in f for t in fi.atoms(TrigonometricFunction)}
+    add, other = sift(targs, lambda x: x.args[0].is_Add, binary=True)
+    trep = {}
+    for i in add:
+        a = i.args[0]
+        if len(a.args) == 2:
+            ind, dep = a.as_independent(*symbols)
+            if dep in symbols or -dep in symbols:
+                # don't let expansion expand wrt anything in ind
+                n = Dummy() if not ind.is_Number else ind
+                trep[i] = TR10(i.xreplace({ind: n})).xreplace({n: ind})
```

Comment:
Is the `xreplace` still needed if only `TR10` is being used?

Reply:
```python
>>> TR10(cos(x+z)).expand()
-sin(x)*sin(z) + cos(x)*cos(z)
>>> TR10(cos(x+20*y+100*z)).expand()
-sin(x)*sin(20*y)*cos(100*z) - sin(x)*sin(100*z)*cos(20*y) - sin(20*y)*sin(100*z
)*cos(x) + cos(x)*cos(20*y)*cos(100*z)
```

Reply:
Yes, to avoid hollow expansion.